core/templates/dev/head/pages/exploration_editor/editor_tab/StateHintsDirective.js

```diff
@@ -0,0 +1,259 @@
+// Copyright 2017 The Oppia Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Controllers, services and filters for hints.
```

Comment:
Update fileoverview -- it's just a single directive for the hint editor, right?

Perhaps should be called stateHintsEditor (for consistency with stateInteractionEditor etc.)

Reply:
Actually, there is a separate hint editor directive. This is just responsible for displaying the hints list, take care of Add Hint and call the hint editor. Should I rename to something else?

Reply:
I think stateHintsEditor is still fine, as long as you keep the plural in the name.

Reply:
Individual hint directives can just be hintEditor.

Reply:
Ok, done.